### PR TITLE
Dls 10863/email validation fix

### PR DESCRIPTION
**Pull request by vamsikrishnapujari**

### CODE REVIEW

File: app/uk/gov/hmrc/cbcrfrontend/emailaddress/EmailAddress.scala [hunk 1]
```diff
@@ -54,43 +31,11 @@ trait EmailValidation {
...
-      case _ =>
-        getAttributeValue(domain, "A") match {
-          case Right(value) => value.nonEmpty
-          case Left(_) => false
-        }
+  def isValid(email: String): Boolean = {
+    val emailConstraint: Constraint[String] = Constraints.emailAddress
+    emailConstraint(email) match {
+      case Valid => true
+      case _     => false
```
Comment (ruthdas03) [inline app/uk/gov/hmrc/cbcrfrontend/emailaddress/EmailAddress.scala:38]: The `false` case is actually `Invalid` and we should be showing it as an error.

File: test/uk/gov/hmrc/cbcrfrontend/emailaddress/EmailAddressSpec.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class EmailAddressSpec extends AnyWordSpec with Matchers {
+  "EmailValidation" should {
+    "validate a correct email address" in {
+      new EmailAddressValidation().isValid("user@test.com") shouldBe true
+      new EmailAddressValidation().isValid("user@test.co.uk") shouldBe true
+      new EmailAddressValidation().isValid("a@a") shouldBe true
+    }
+    "validate invalid email" in {
+      new EmailAddressValidation().isValid("user @test.com") shouldBe false
+      new EmailAddressValidation().isValid("user@") shouldBe false
```
Comment (ruthdas03) [inline test/uk/gov/hmrc/cbcrfrontend/emailaddress/EmailAddressSpec.scala:31]: Incase of `false` there should be some error

**Review by ruthdas03 — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/cbcrfrontend/emailaddress/EmailAddress.scala (hunk 1 above)
Comment (vamsikrishnapujari) [reply app/uk/gov/hmrc/cbcrfrontend/emailaddress/EmailAddress.scala]: Changed false case to log invalid email.

**Review by ruthdas03 — APPROVED** [review]
please note: ATs failing for me due to local env issues

**Pull request merged**